SalesforceSDK/Salesforce.SDK.Core/Source/Auth/AuthStorageHelper.cs

```diff
@@ -420,5 +421,94 @@ internal void DeleteData(string key)
                 _persistedData.Values.Remove(key);
             }
         }
+
+        internal void PersistEncryptionSettings(string password, string salt, string nonce = null)
+        {
+            DeleteEncryptionSettings();
+            var encryptionSettingsObj = new { Password = password, Salt = salt };
+            var encrpytionSettings = new PasswordCredential(PasswordVaultSecuredData, PasswordVaultEncryptionSettings,
+                JsonConvert.SerializeObject(encryptionSettingsObj));
+            _vault.Add(encrpytionSettings);
+            PlatformAdapter.SendToCustomLogger("AuthStorageHelper.PersistEncryptionSettings - encryption settings added to vault",
+                LoggingLevel.Verbose);
+        }
+
+        internal bool TryRetrieveEncryptionSettings(out string password, out string salt, string nonce = null)
+        {
+            password = null;
+            salt = null;
+            PasswordCredential creds = SafeRetrieveResource(PasswordVaultSecuredData).FirstOrDefault();
+            if (creds != null)
+            {
+                PasswordCredential encrpytionSettings = _vault.Retrieve(PasswordVaultSecuredData, PasswordVaultEncryptionSettings);
+                if (String.IsNullOrWhiteSpace(encrpytionSettings.Password))
+                {
+                    // Failed to deserialize the data, we should clear it out and start over.
+                    PlatformAdapter.SendToCustomLogger(
+                        "AuthStorageHelper.TryRetrieveEncryptionSettings - Encryption Settings values are corrupt. Assuming bad state and clearing the vault completely",
+                        LoggingLevel.Verbose);
+                    _vault.Remove(encrpytionSettings);
+                    DeletePersistedCredentials();
+                    DeletePincode();
+                }
+                else
+                {
+                    try
+                    {
+                        var encrpytionSettingsObj = JsonConvert.DeserializeObject<dynamic>(encrpytionSettings.Password);
+                        password = encrpytionSettingsObj.Password;
+                        salt = encrpytionSettingsObj.Salt;
+                        PlatformAdapter.SendToCustomLogger(
+                        "AuthStorageHelper.TryRetrieveEncryptionSettings - Encryption Settings have been retrieved successfully.",
+                        LoggingLevel.Verbose);
+                        return true;
+                    }
+                    catch (Exception ex)
+                    {
+                        // Failed to deserialize the data, we should clear it out and start over.
+                        PlatformAdapter.SendToCustomLogger(
+                            "AuthStorageHelper.TryRetrieveEncryptionSettings - Encryption Settings values can't be deserialized. Assuming bad state and clearing the vault completely",
+                            LoggingLevel.Verbose);
```

Comment:
+1 on Wes's comment. If an exception happens it either should be logged at Critical level (in case it cannot be ignored) or at Warning level (in case it can be ignored, like in here).